### ISSUE TITLE
Shop doesn't differentiate between item ID 5:0 and 5:1

### ISSUE THREAD

**Issue by aWitch-Doctor** [TheoKah/CarrotShop#6]
You can do this for all items that have a sub ID. stone (id: 1:0) and andesite (id: 1:6), either can be sold to a shop that was created with only one of those. Same with potions and enchanted books and every other item that has a sub ID. The items stay different in the chest. I could sell eff 1 enchanted book to a shop that is buying silk touch books. If i did this and opened the chest I would see 2 different books, 1 eff 1 and 1 silk.

I'm using Sponge Vanilla API 5 build 385, and total economy and a bunch more but I don't think those affect this.

**Comment by TheoKah** [comment]
Problem with stone and wool is fixed in last version.
I'm aware of the problem with enchanted stuff. This is due to a problem in the deserialisation of itemstack by sponge.

I need to dig a bit more as they claim they fixed the problem already

**Comment by aWitch-Doctor** [comment]
well damn, when i clicked the versions link on the sponge site i saw 1.3 as the latest. Didn't think to look here for a newer version. Unfortunately I still can't use this until it works for all items. I'll keep checking back. Thank you for your hard work on this plugin.

**Comment by TheoKah** [comment]
Long story short. I've been away from my main computer which has the key needed to sign the newest version so I can upload it on sponge.

This bug is my priority. But might take some time (matter of days)

**Comment by aWitch-Doctor** [comment]
Wow, I can live with days. The other plugin i tried to use for shops didn't really tell me much. I figured it would be months waiting for sponge to update. Then another few weeks waiting on that creator to update his plugin. Thank you again for all your work on this.

**Comment by TheoKah** [comment]
This should be fixed with v1.5

Let me know if the problem you identified is fixed
https://github.com/TheoKah/CarrotShop/releases/tag/1.5

**Comment by aWitch-Doctor** [comment]
You have fixed the enchanted books. I only tried to sell 1 book to a shop made with a different book, i'll test this more later. 

Potions still do this though. I made a shop to buy strength 2 potions from the player and I was able to sell Strength 1 potions, both 3 and 8 minute.

Also a few requests if it isn't too much work. 1 and 2 are the ones i would really like to see, the others are extra.

1. Can you make this work with trapped chests as well? 
2. Are you able to add permissions for creating buy, sell, and trade shops? I want to restrict what world they can make these shops in and that is the easiest way for me to do it.
3. Any way to change this to see both chests of a double chest for both regular and trapped chest? If the player puts down a double chest then the shop only sees half the chest and the items must be in that half, as soon as that half is full then it stops buying. Same for selling, as soon as that half is empty it stops selling.
4. Any chance of being able to have both a buy and sell sign linked to the same chest?

**Comment by TheoKah** [comment]
I'll fix if for the potions too, forgot about that.

1. I'll see what I can do, but I think I can do that. Maybe even use OreDict as said in #2 

2. I'll add perms for those, yes.

3. About the double chests, I need to check what sponge has to offer before I can answer for sure.

4. This is probably something I can do, but this is not a priority for now.

**Comment by aWitch-Doctor** [comment]
YAY! thank you, If you can add trapped chests then double chests aren't really needed, would just be nice. Linking buy and sell to one chest has a couple options, a new sign that has both buy and sell. Line 1: [Buy:Sell] Line 4 10:5 (buy price:sell price) then left click is buy right click is sell or 2 different signs linking to the same chest. Again, that is not really needed, just a thought.

Thank you for doing all this work. I really appreciate it.

**Comment by aWitch-Doctor** [comment]
Lol, had my first issue trying to show someone the plugin. Can you make it so when a player clicks a sign it says in their chat "(Sold/Bought) (quantity) (item) for (price)". Right now it's not saying anything so they spammed it yelling at me asking why it wasn't working.

Also, not sure if this is anything but i figured i would ask. When you punch the sign it says what the sign does. But it has something weird for the name. "Trade item.dyePowder.name x1 and get Golden Apple x1?"
same for stone, it shows tile.stone.name x64.
https://puu.sh/weutk/2693387fb0.png

And I was just told that spawn eggs and arrows are doing the same thing as potions now. Set up a shop for one and you can buy or sell any of them to that shop.

**Comment by aWitch-Doctor** [comment]
The plugin does not like decimals... 0.1 would be 1 dollar. 1.5 would be 15 dollars. any chance this can be changed?

**Comment by TheoKah** [comment]
I'll check that for the names

About the confirm message, I added them already but lost the commit when I switched pc. You will have it back in next version (probably tomorrow)

I'll add the eggs and arrow. I have to do those special items manually because if I just generalise, this breaks compatibility with some modded items. But I want to have it working for 100% of vanilla items.

**Comment by TheoKah** [comment]
About the decimal. This is not something I want to add now. I had problems when I first tried to handle that, and you can achieve similar goal changing the number or items.

Instead of selling 1 item for 0.1, you will sell 10 for 1

**Comment by aWitch-Doctor** [comment]
Not a problem about the decimal, just wanted to make sure you were aware. 

I just thought about it, enchanted items like shovel, pickaxe, etc. do the same as the eggs and arrows, not sure how that will work out with being able to rename them and enchants being able to have different orders. I don't think this will really be an issue since people would sell these items in their shop and not try to buy them. Enchanted books have to have the same enchants on them in the same order for the shop to allow it. The plugin doesn't care about names of items which is good.

**Comment by TheoKah** [comment]
https://github.com/TheoKah/CarrotShop/releases/tag/1.6

Added most things we talked about. As the initial issue is now fixed, I'll close this one.
Feel free to open a new one if you find another problem or have any suggestion
